snuba/query/snql/parser.py

```diff
@@ -38,18 +61,21 @@
 
 snql_grammar = Grammar(
     r"""
-    query_exp             = match_clause where_clause? collect_clause? group_by_clause? having_clause? order_by_clause?
+    query_exp             = match_clause where_clause? collect_clause? group_by_clause? having_clause? order_by_clause? limit_clause?
 
-    match_clause          = space* "MATCH" space* open_paren clause close_paren space*
+    match_clause          = space* "MATCH" space* entity_match space*
```

Comment:
I cannot find the comment on the old PR, so I cannot remember where we landed on this.  I think we should impose at least one space between keywords and identifiers, so `space+` otherwise unreadable queries like: `MATCHeventsCOLLECTevent_it` would be valid. Which, besides being very hard to read, I suspect will end up being ambiguous in some cases.

Reply:
I had this in the other PR, so when I merge that other PR the spacing changes will be there. It's not a difficult change to make.